modin/pandas/test/test_series.py

```diff
@@ -1620,8 +1620,14 @@ def test_dtype(data):
     df_equals(modin_series.dtype, pandas_series.dtypes)
 
 
-def test_dt():
-    data = pd.date_range("2016-12-31", periods=128, freq="D", tz="Europe/Berlin")
+# Bug https://github.com/modin-project/modin/issues/4436 in
+# Series.dt.to_pydatetime is only reproducible when the date range out of which
+# the frame is created has timezone None, so that its dtype is datetime64[ns]
+# as opposed to, e.g. datetime64[ns, Europe/Berlin]. To reproduce that bug, we
+# use timezones None and Europe/Berlin.
+@pytest.mark.parametrize("timezone", [None, "Europe/Berlin"])
+def test_dt(timezone):
+    data = pd.date_range("2016-12-31", periods=128, freq="D", tz=timezone)
     modin_series = pd.Series(data)
```

Comment:
I think the tests need to be modified to actually ensure that the dtypes are correct. As I mentioned before, `df_equals` doesn't check dtypes by default, so even if we have a clause in the test that should be triggering the bug, we don't actually confirm that the bugfix works since we don't check the output dtypes.

Reply:
That makes sense, thank you!